software/src/modules/ac011k/ac011k.cpp

```diff
@@ -1284,11 +1284,41 @@ void AC011K::loop()
                     meter.updateMeterAllValues(METER_ALL_VALUES_CURRENT_L3_A, float(PrivCommRxBuffer[110]+256*PrivCommRxBuffer[111])/10);
 
                     // meter power
+                    static float prev_energy_rel_raw;
+                    static float prev_energy_abs_raw;
+                    float energy_rel_raw = float(PrivCommRxBuffer[84]+256*PrivCommRxBuffer[85])/1000;
+                    float energy_abs_raw = float(PrivCommRxBuffer[88]+256*PrivCommRxBuffer[89])/1000;
+
+                    // detect meter restart
+                    if (energy_rel_raw < ac011k_hardware.meter.get("energy_rel_raw")->asFloat()) { ac011k_hardware.meter.get("energy_rel_delta")->updateFloat(ac011k_hardware.meter.get("energy_rel")->asFloat()); }
+                    if (energy_abs_raw < ac011k_hardware.meter.get("energy_abs_raw")->asFloat()) { ac011k_hardware.meter.get("energy_abs_delta")->updateFloat(ac011k_hardware.meter.get("energy_abs")->asFloat()); }
+
+                    // persist meter on change
+                    if ((prev_energy_rel_raw != energy_rel_raw) || (prev_energy_abs_raw != energy_abs_raw)) {
+                        if(ac011k_hardware.config.get("verbose_communication")->asBool()) {
+                            logger.printfln(" energy_rel: %.1fWh, energy_rel_raw: %.1fWh, energy_rel_delta: %.1fWh, energy_abs: %.1fWh, energy_abs_raw: %.1fWh, energy_abs_delta: %.1fWh", 
+                                1000 * ac011k_hardware.meter.get("energy_rel")->asFloat(),
+                                1000 * ac011k_hardware.meter.get("energy_rel_raw")->asFloat(),
+                                1000 * ac011k_hardware.meter.get("energy_rel_delta")->asFloat(),
+                                1000 * ac011k_hardware.meter.get("energy_abs")->asFloat(),
+                                1000 * ac011k_hardware.meter.get("energy_abs_raw")->asFloat(),
+                                1000 * ac011k_hardware.meter.get("energy_abs_delta")->asFloat());
+                        }
+                        prev_energy_rel_raw = energy_rel_raw;
+                        prev_energy_abs_raw = energy_abs_raw;
+                        api.writeConfig("ac011k_hardware/meter", &ac011k_hardware.meter);
```

Comment:
amount of writes to flash maybe to high. Write every hour?

Reply:
done. once per hour and on finished charging.